subsys/bluetooth/mesh/adv_ext.c

```diff
@@ -103,9 +102,12 @@ static int adv_start(const struct bt_le_adv_param *param,
 	if (err) {
 		BT_ERR("Advertising failed: err %d", err);
 		atomic_clear_bit(adv.flags, ADV_FLAG_ACTIVE);
+		return err;
 	}
 
-	return err;
+	net_buf_ref(adv.buf);
```

Comment:
No code should ever call `net_buf_ref()` without storing the return value to a new pointer. See my other comment, i.e. the refcounting will need some redesign.

Reply:
@trond-snekvik Any opinion to this ?

Reply:
I agree with @jhedberg :)